### PR TITLE
Add support for services

### PR DESCRIPTION
**Pull request by marcelveldt**
Add (basic) support for services, to be extended later on.

I've decided to move the services stuff into a separate file/class for code hygiene/readability reasons, in the old implementation this was all packed into the config_flow method which is horror to read/follow.

### CODE REVIEW

File: custom_components/zwave_mqtt/services.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        })
+        add_node_schema = vol.Schema({
+            vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+            vol.Optional(const.ATTR_SECURE, default=False): vol.Coerce(bool),
+        })
+        node_schema = vol.Schema({
+            vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+            vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+        })
+        set_config_parameter_schema = vol.Schema({
```
Comment (balloob) [inline custom_components/zwave_mqtt/services.py]: Can you skip making them variables, just pass `schema=vol.Schema`. That's the direction of HA.

File: custom_components/zwave_mqtt/services.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        self._hass.services.async_register(const.DOMAIN,
+                                           const.SERVICE_CANCEL_COMMAND,
+                                           self.cancel_command,
+                                           schema=instance_schema)
+        self._hass.services.async_register(const.DOMAIN,
+                                           const.SERVICE_SET_CONFIG_PARAMETER,
+                                           self.set_config_parameter,
+                                           schema=set_config_parameter_schema)
+
+    def add_node(self, service):
```
Comment (balloob) [inline custom_components/zwave_mqtt/services.py:92]: Each of these functions need to be decorated with `@callback`

File: custom_components/zwave_mqtt/services.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            # List value
+            if value.type == const.TYPE_LIST:
+                return value.send_value(str(selection))
+            # Button
+            if value.type == const.TYPE_BUTTON:
+                value.send_value(True)
+                value.send_value(False)
+                return
+            # Byte value
+            return value.send_value(int(selection))
```
Comment (balloob) [inline custom_components/zwave_mqtt/services.py]: This function doesn't return anything. Put return on a new line.

File: custom_components/zwave_mqtt/services.py (hunk 1 above)
Comment (marcelveldt) [reply custom_components/zwave_mqtt/services.py]: makes sense, thanks for explaining. will adjusty

File: custom_components/zwave_mqtt/const.py [hunk 4]
```diff
@@ -16,6 +16,20 @@
 # Entity Attributes
```
Comment (balloob) [inline custom_components/zwave_mqtt/const.py:16]: Please undo the file permissions mod. (644->755)

**Comment by marcelveldt** [comment]
All fixed. No idea where the linter in Travis is angry about. Both flake8 and pylinter are happy on my setup with default PEP8 style conventions enabled.

**Comment by balloob** [comment]
Both isort and black are not satisfied: https://travis-ci.com/cgarwood/homeassistant-zwave_mqtt/builds/145290764

Locally run `pip install pre-commit` and then `pre-commit install` and it will fix it for you when making commits 👍 

For now manually run `isort` and `black` on the files and commit changes.

**Review by balloob — APPROVED** [review]
Ok to merge when travis passes.

**Comment by balloob** [comment]
btw for Travis, you can click on "Details" and then click on "The Build"

**Comment by balloob** [comment]
Are you on Discord? Let's connect there.

**Pull request merged**